File: src/main/java/mediathek/tool/cellrenderer/CellRendererPset.java
package mediathek.tool.cellrenderer;

import mediathek.daten.DatenPset;
import mediathek.tool.SVGIconUtilities;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import javax.swing.*;
import javax.swing.table.DefaultTableCellRenderer;
import java.awt.*;

public class CellRendererPset extends DefaultTableCellRenderer {
    private static final ImageIcon ja_16 = SVGIconUtilities.createSVGIcon("icons/fontawesome/check.svg");
    private static final Logger logger = LogManager.getLogger();

    public CellRendererPset() {
    }

    @Override
    public Component getTableCellRendererComponent(
            JTable table,
            Object value,
            boolean isSelected,
            boolean hasFocus,
            int row,
            int column) {
        setBackground(null);
        setForeground(null);
        setFont(null);
        setIcon(null);
        setHorizontalAlignment(SwingConstants.LEADING);
        super.getTableCellRendererComponent(
                table, value, isSelected, hasFocus, row, column);
        try {
            int r = table.convertRowIndexToModel(row);
            int c = table.convertColumnIndexToModel(column);
            DatenPset datenPset = new DatenPset();
            for (int i = 0; i < DatenPset.MAX_ELEM; ++i) {
                datenPset.arr[i] = table.getModel().getValueAt(r, i).toString();
            }
            if (c == DatenPset.PROGRAMMSET_NAME) {
                setForeground(datenPset.getFarbe());
            }
            if (c == DatenPset.PROGRAMMSET_IST_ABSPIELEN) {
                setHorizontalAlignment(SwingConstants.CENTER);
                setText(""); // nur das Icon anzeigen
                if (datenPset.istAbspielen()) {
                    setIcon(ja_16);
                } else {
                    setIcon(null);
                }
            }
            if (c == DatenPset.PROGRAMMSET_IST_SPEICHERN) {
                setHorizontalAlignment(SwingConstants.CENTER);
                setText(""); // nur das Icon anzeigen
                if (datenPset.istSpeichern()) {
                    setIcon(ja_16);
                } else {
                    setIcon(null);
                }
            }
        } catch (Exception ex) {
            logger.error("getTableCellRendererComponent", ex);
        }
        return this;
    }
}

File: src/main/kotlin/mediathek/tool/cellrenderer/CellRendererPset.kt
package mediathek.tool.cellrenderer

import mediathek.daten.DatenPset
import mediathek.tool.SVGIconUtilities
import org.apache.logging.log4j.LogManager
import java.awt.Component
import javax.swing.JTable
import javax.swing.SwingConstants
import javax.swing.table.DefaultTableCellRenderer

class CellRendererPset : DefaultTableCellRenderer() {
    override fun getTableCellRendererComponent(
        table: JTable,
        value: Any?,
        isSelected: Boolean,
        hasFocus: Boolean,
        row: Int,
        column: Int,
    ): Component {
        background = null
        foreground = null
        font = null
        icon = null
        horizontalAlignment = SwingConstants.LEADING
        super.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, column)
        try {
            val modelRow = table.convertRowIndexToModel(row)
            val modelColumn = table.convertColumnIndexToModel(column)
            val datenPset = DatenPset()
            for (i in 0 until DatenPset.MAX_ELEM) {
                datenPset[i] = table.model.getValueAt(modelRow, i).toString()
            }
            if (modelColumn == DatenPset.PROGRAMMSET_NAME) {
                foreground = datenPset.farbe
            }
            if (modelColumn == DatenPset.PROGRAMMSET_IST_ABSPIELEN) {
                drawBooleanIcon(datenPset.istAbspielen())
            }
            if (modelColumn == DatenPset.PROGRAMMSET_IST_SPEICHERN) {
                drawBooleanIcon(datenPset.istSpeichern())
            }
        } catch (ex: Exception) {
            logger.error("getTableCellRendererComponent", ex)
        }
        return this
    }

    private fun drawBooleanIcon(enabled: Boolean) {
        horizontalAlignment = SwingConstants.CENTER
        text = ""
        icon = if (enabled) checkIcon else null
    }

    private companion object {
        private val checkIcon = SVGIconUtilities.createSVGIcon("icons/fontawesome/check.svg")
        private val logger = LogManager.getLogger()
    }
}
